src/Microsoft.VisualStudio.LanguageServices.Razor/VisualStudioRazorParser.cs

```diff
@@ -0,0 +1,151 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.AspNetCore.Razor.Language;
+using Microsoft.AspNetCore.Razor.Language.Legacy;
+using Microsoft.VisualStudio.Text;
+using ITextBuffer = Microsoft.VisualStudio.Text.ITextBuffer;
+
+namespace Microsoft.VisualStudio.LanguageServices.Razor
+{
+    public class VisualStudioRazorParser : IDisposable
```

Comment:
- Lifetime is going to be managed [in a follow up](https://github.com/aspnet/Razor/issues/1630).
- Thinking more about abstracting this, that might be overkill, at least for now. We currently have multi-targeting support at the template engine level; do you also envision us needing to change how we parse the Razor language syntax based on the Razor targeted by the project?